src/hooks/useQuery.ts

```diff
@@ -63,7 +65,14 @@ export const useQuery = <T = any, V = object>(
         })
       );
     },
-    [args.requestPolicy, client, devtoolsContext, request, setState]
+    [
+      args.operationContext,
```

Comment:
I think this is potentially a huge problem. I think people may accidentally misunderstand the context and cause too many refetches. This does mean that the rest spread may be the worse option and `args.context` the better one, but it's still something we should keep in mind

Reply:
So rename `operationContext` --> context and spread it out in the args array? Since that would enable our users to not having to memoize creating an object in useMemo argument?

Reply:
I think spreading isn't a good idea, yea, as said it'd make it even harder to make sure the reference doesn't change. It was something we had at some point, I believe, but removed it. Maybe it was even due to this problem

Reply:
I remember us having a similar problem at one of my former projects, we ended up making a hook `usePreviousValue` and deeply comparing them. I would suggest leaving args.context out of the array to be honest.... This is against the rules of hooks but in theory it doesn't impact this concept of ours

Reply:
Yea, I don't think we should go down that road; maybe let's just add "args.context" for now and not worry about this too much?

Reply:
That would make me having to adjust the tests though, I'm not too fond about that

Reply:
I pushed a fix for the failing test, should something like this fix it:

```
const context = React.useMemo(() => args.context, [...args.context]);
```

This however imposes an issue with `fetchOptions`

Reply:
FWIW I had to deal with this same problem in my hook that I described in #346.  Deciding that users of my hook (who are supplying values that go into the context) were very unlikely to provide referentially equal values across renders, I excluded those values from the react hook dependencies and added a lint to force my hook to be used with constant values 🙃.

Obviously you are dealing with a much more general problem than I am so this may still be the right solution, but for my use case I will have to keep pausing and invoking executeQuery myself to get around this dependencies check.